### PR TITLE
Electric Motor volume weight and power adjustement

### PR DESCRIPTION
**Pull request by Fris0uman**
#### Summary
```SUMMARY: Balance "Changes volume, weight and power of electric motors to real life values"```

#### Purpose of change
The rework of vehicle physic by mlangsdorf allows to make electric motor match some real life counterpart in volume power and weight without breaking game balance.

#### Describe the solution
Implements values that were sugessted in #27335
Most power were already set at the right values.
I also introduced the super electric motor, which is the most powerful motor but also the biggest. It only spawns in Electric racing car.

#### Additional context

Values as suggested by @mlangsdorf  in #27335

| Motor Size | Power (kW) | Volume (L) | Mass (kg) | Reference |
| :---------- | :----------: | :----------: | :----------: | :----------|
| Micro Electric Motor |0.1 |0.25| 0.4| No reference|
| Tiny Electric Motor |	0.75 	|0.5| 	5.8| 	https://www.ebikekit.com/collections/wheels/products/front-500w-heavy-duty-direct-drive-motor
| Small Electric Motor 	|7.3| 	5| 	7.5| 	https://www.miromax.lt/en/m-6/c-39/c-45-brushless_bldc_motor_with_permanent_magnet/product-500-bldc__pmsm_brushless_motor_hpm-5000b_-_nominal_power_5kw86kw__67hp115hp__400_cm3#to_products
 |Electric Motor| 	37.3 |	9.7| 	27.2| 	http://www.evwest.com/catalog/product_info.php?cPath=8&products_id=339
| "Obsolete" Large Electric Motor| 	93.25 |	20.2| 	68| 	http://www.evwest.com/catalog/product_info.php?cPath=8&products_id=170
| Large Electric Motor| 	149.2 |	17.4 	|68| 	http://www.evwest.com/catalog/product_info.php?cPath=8&products_id=297
| Enhanced Electric Motor 	|313.3 	|26.5 	|81.6 	|http://www.evwest.com/catalog/product_info.php?products_id=300
| Super Electric Motor| 	400 	|111 	|133.8 	|http://www.evwest.com/catalog/product_info.php?products_id=476

The Electric Racing car is an electric sports car with the 2 enhanced motors replaced by one super motor, it can stay as is or I can remove it and just replace the enhanced motor in the atomic sports car to avoid making a new vehicle.

Also should I add the super motor to an item group and if so which one ?

### CODE REVIEW

**Comment by KITbutler** [comment]
Your stats don't align with the sources: the "enhanced motor" reference is 16l case volume not 26+l and the Tesla motor is 167l, the masses seem to be close enough for both. However for both the power output compared to their electricity consumption violates conservation of energy, something the recent change to the current stats explicitly aimed to rectify. 

Given the distinct lack of RL nuclear-powered sports cars and scarcity of solar options, is there any *gameplay* benefit to these changes? The current weight values are too low but the near-future tech shown in game suggests some improvements would not be impossible, for example the copper used in most coils nowadays is almost four times as dense as silicon which forms the base substrate for many semiconductor devices.

**Comment by Fris0uman** [comment]
>violates conservation of energy

Yep just fixed it, effienciency set at 89.9% as it seems to be what large motor has.

>The current weight values are too low

It seems to be the case, and the Pr aims to fixe that.

>But near future tech

I can't realy object anything to that, if it doesn't fit the PR will be rejected.

>gameplay

Probably not many great benefit no, but probably no great penalty either.

**Comment by KITbutler** [comment]
> Probably not many great benefit no, but probably no great penalty either.

The vastly increased weight is a flat-out nerf especially to small vehicles like e-bikes so my question basically boils down to whether an intermediate value would be *better* for gameplay.

The volume of the Tesla motor is somewhat important because my calculation puts it above the maximum for a folding shopping cart but that was for a bounding box, going by the picture it might be closer to a 12in diameter cylinder plus a 12in cube which equates to 94l. So I guess your value is okay since volume really doesn't matter much for vehicle parts.

**Comment by Fris0uman** [comment]
>an intermediate value would be better

So as far as I understands how things work on this project you are very welcome to suggest other values and back them up with references.
I picked up mlangsdorf 's suggestion because it was well documented and kind of closed the discussion on the subject.
If you find references to intermediate values,  i'll be happy to use them.

File: data/json/vehicleparts/motor.json [hunk 1]
```diff
@@ -23,7 +23,7 @@
     "looks_like": "motor_small",
     "difficulty": 1,
     "durability": 80,
-    "power": 1492,
+    "power": 750,
     "energy_consumption": 1750,
```
Comment (mlangsdorf) [inline data/json/vehicleparts/motor.json]: Why did this motor suddenly become half as efficient?

File: data/json/vehicles/cars.json [hunk 2]
```diff
@@ -713,6 +713,81 @@
...
+    ],
+    "parts": [
+      { "x": 0, "y": 0, "part": "frame_vertical_2" },
+      { "x": 0, "y": 0, "part": "seat" },
+      { "x": 0, "y": 0, "part": "seatbelt" },
+      { "x": 0, "y": 0, "part": "controls" },
+      { "x": 0, "y": 0, "part": "dashboard" },
+      { "x": 0, "y": 0, "part": "vehicle_alarm" },
+      { "x": 0, "y": 0, "part": "horn_car" },
+      { "x": 0, "y": 0, "part": "roof" },
```
Comment (mlangsdorf) [inline data/json/vehicles/cars.json]:
```suggestion
      { "x": 0, "y": 0, "parts": [ "xlframe_vertical_2", "seat", "seatbelt", "controls", "dashboard", "vehicle_alarm", "horn_car", "roof" ] },
```
Please use the parts array for a more compact listing.

File: data/json/vehicles/cars.json [hunk 3]
```diff
@@ -713,6 +713,81 @@
...
+      { "x": -2, "y": -1, "part": "xlframe_horizontal" },
+      { "x": -2, "y": -1, "part": "xlhalfboard_sw" },
+      { "x": -2, "y": -1, "part": "wheel_slick_steerable" },
+      { "x": -2, "y": 0, "part": "xlframe_horizontal" },
+      { "x": -2, "y": 0, "part": "door_trunk" },
+      { "x": -2, "y": 1, "part": "xlframe_horizontal" },
+      { "x": -2, "y": 1, "part": "door_trunk" },
+      { "x": -2, "y": 2, "part": "xlframe_horizontal" },
+      { "x": -2, "y": 2, "part": "xlhalfboard_se" },
+      { "x": -2, "y": 2, "part": "wheel_slick_steerable" }
```
Comment (mlangsdorf) [inline data/json/vehicles/cars.json]: Nice!  Good use of slick wheels.

But not all the of the frames are xlframes, and a racing car should only have extra-light frames.

File: data/json/items/vehicle_parts.json [hunk 4]
```diff
@@ -336,11 +350,11 @@
         "id" : "motor_small",
         "name" : "small electric motor",
         "description" : "A small electric motor.  Useful for crafting.",
-        "weight" : 1500,
+        "weight" : 7500,
         "color" : "light_cyan",
         "symbol" : ",",
         "material" : ["steel"],
-        "volume" : 2,
+        "volume" : 20,
```
Comment (mlangsdorf) [inline data/json/items/vehicle_parts.json]: Make this in "5 L" or whatever for consistency.

File: data/json/vehicleparts/motor.json (hunk 1 above)
Comment (Fris0uman) [reply data/json/vehicleparts/motor.json]: My bad, forgot to adjust energy_consumption.

File: data/json/vehicles/cars.json (hunk 3 above)
Comment (Fris0uman) [reply data/json/vehicles/cars.json]: It was originally a literal copy-paste of electric sports car where I just changed the motor to use the super motor. Which means another PR might need to change sports car to have only xlframes.

I implemented your suggestions, but I have one question what's the point of having only steerable wheels, shouldn't we have only steerable wheel at the front ?

**Comment by kevingranade** [comment]
The vehicle efficiency tests need to be tweaked to pass after this change.

**Comment by Fris0uman** [comment]
I must admit that I have no idea how to do that, so I would prefer if someone else took care of it.

**Comment by kevingranade** [comment]
That's fine, I can handle it as part of the merge.

**Comment by kevingranade** [comment]
Merged into 0.D and master.

**Pull request closed without merging**